xdsl/ir/core.py

```diff
@@ -536,21 +536,52 @@ class Use:
     """The index of the operand using the value in the operation."""
 
 
+@dataclass
+class IRUses(Iterable[Use]):
+    """
+    Multi-pass iterable of the uses of an IR value (SSAValue or Block).
+    """
+
+    ir: IRWithUses
```

Comment:
Maybe it will become more apparent in further PRs but why take the IR here instead of the set.

Reply:
If it's going to disappear anyway then this seems fine